### PR TITLE
Predict Operator maturity support for OM 1.7

### PR DESCRIPTION
**Pull request by karol-czarnecki**
This PR updates predict playbooks with following:
 
- OM 1.7 - add feature to ansible-devops predict playbook #13605
~~- OM 3.1: Workload Scale Customization (ibm-mas-predict) #13041~~
- changes to CRD's thats part of MAS 8.10

### CODE REVIEW

**Review by lokesh-sreedhara — DISMISSED** [review]

**Review by lokesh-sreedhara — DISMISSED** [review]

**Review by lokesh-sreedhara — APPROVED** [review]

**Comment by lokesh-sreedhara** [comment]
@whitfiea yea, this PR is still valid and needs to be merged. @karol-czarnecki removed the OM 3.1 changes from this PR a while back. 

its left with only OM 1.7. Let me rename the PR title.

**Comment by whitfiea** [comment]
@lokesh-sreedhara what about things like this https://github.com/ibm-mas/ansible-devops/pull/668/files#diff-6a3d629a646a90bd6eb3cfea5207336ec7722d469b89a942f7e947b7b8d6aeceR60 the mention of the WORKLOAD_SCALING_CONFIGMAP? and other changes in that file?

**Comment by karol-czarnecki** [comment]
Hi @whitfiea 

I can review and refactor this PR if really all was removed from OM 3.1 - tomorrow morning, I have some other task now.

**Comment by karol-czarnecki** [comment]
Hi @whitfiea 

Removed this part from predict documentation. Sorry about that. Now PR is ready to merge

File: ibm/mas_devops/roles/suite_app_install/vars/defaultspecs/predict.yml [hunk 1]
```diff
@@ -1,10 +1,7 @@
 ---
 # Default application specs for Predict
 mas_app_spec:
-  bindings:
-    jdbc: "{{ mas_app_bindings_jdbc }}"
   components: {}
```
Comment (whitfiea) [inline ibm/mas_devops/roles/suite_app_install/vars/defaultspecs/predict.yml:4]: the jdbc binding is not needed anymore?

File: ibm/mas_devops/roles/suite_app_install/vars/defaultspecs/predict.yml [hunk 2]
```diff
@@ -1,10 +1,7 @@
...
   settings:
-    deployment:
-      size: "{{ mas_app_settings_predict_deployment_size}}"
```
Comment (whitfiea) [inline ibm/mas_devops/roles/suite_app_install/vars/defaultspecs/predict.yml:9]: If we aren't going workload scaling then we still need the deployment size don't we?

File: ibm/mas_devops/roles/suite_app_config/vars/predict.yml [hunk 3]
```diff
@@ -5,3 +5,17 @@ mas_app_ws_kind: PredictWorkspace
 
 mas_app_cfg_delay: 120
 mas_app_cfg_retries: 30
+
+mas_app_settings_wml_instance_id: "{{ lookup('env', 'CPD_WML_INSTANCE_ID') | default('openshift', true) }}"
+mas_app_settings_wml_url: "{{ lookup('env', 'CPD_WML_URL') | default('https://internal-nginx-svc.ibm-cpd.svc:12443', true) }}"
+mas_app_settings_wml_version: "{{ lookup('env', 'WML_VERSION') | default('4.5', true) }}"
+mas_app_settings_watsonstudio_project_id: "{{ lookup('env', 'CPD_WSL_PROJECT_ID') | default('', true) }}"
```
Comment (whitfiea) [inline ibm/mas_devops/roles/suite_app_config/vars/predict.yml:12]: CPD_WSL_PROJECT_ID was removed from the one_click docs in this same PR. I think this should be CPD_PREDICT_WS_PROJECT?

File: ibm/mas_devops/roles/suite_app_config/vars/predict.yml [hunk 4]
```diff
@@ -5,3 +5,17 @@ mas_app_ws_kind: PredictWorkspace
 
 mas_app_cfg_delay: 120
 mas_app_cfg_retries: 30
+
+mas_app_settings_wml_instance_id: "{{ lookup('env', 'CPD_WML_INSTANCE_ID') | default('openshift', true) }}"
+mas_app_settings_wml_url: "{{ lookup('env', 'CPD_WML_URL') | default('https://internal-nginx-svc.ibm-cpd.svc:12443', true) }}"
```
Comment (whitfiea) [inline ibm/mas_devops/roles/suite_app_config/vars/predict.yml:10]: Are these not the same as what is defined in roles/suite_app_config/defaults/main.yml ?

**Review by whitfiea — CHANGES_REQUESTED** [review]

File: ibm/mas_devops/roles/suite_app_install/vars/defaultspecs/predict.yml (hunk 1 above)
Comment (karol-czarnecki) [reply ibm/mas_devops/roles/suite_app_install/vars/defaultspecs/predict.yml:4]: Yes, it is not needed on main operator. It is needed on ws operator

File: ibm/mas_devops/roles/suite_app_install/vars/defaultspecs/predict.yml (hunk 2 above)
Comment (karol-czarnecki) [reply ibm/mas_devops/roles/suite_app_install/vars/defaultspecs/predict.yml:9]: Hi @whitfiea 
We don't, we disabled this as I implement OM3.1 for MAS 8.10 - but then design has been change. So we don't use this variable any more

File: ibm/mas_devops/roles/suite_app_config/vars/predict.yml (hunk 3 above)
Comment (karol-czarnecki) [reply ibm/mas_devops/roles/suite_app_config/vars/predict.yml:12]: @whitfiea will fix this

File: ibm/mas_devops/roles/suite_app_config/vars/predict.yml (hunk 4 above)
Comment (karol-czarnecki) [reply ibm/mas_devops/roles/suite_app_config/vars/predict.yml:10]: will fix this

**Comment by karol-czarnecki** [comment]
Hi @whitfiea 

Could you please review now?

File: ibm/mas_devops/roles/suite_app_config/vars/predict.yml [hunk 5]
```diff
@@ -5,3 +5,15 @@ mas_app_ws_kind: PredictWorkspace
...
+
+mas_app_settings_wml_version: "{{ lookup('env', 'WML_VERSION') | default('4.5', true) }}"
+mas_app_settings_watsonstudio_project_id: "{{ lookup('env', 'CPD_PREDICT_WS_PROJECT') | default('', true) }}"
+mas_app_settings_imagePullPolicy: "{{ lookup('env', 'MAS_IMAGE_PULL_POLICY') | default('IfNotPresent', true) }}"
+
+# CPD version is required by predict to identify what version of WML must be used to activate predict
+# if this information is used by more than predict, it should be moved to defaults/main.yml of the role
+cpd_product_version: "{{ lookup('env', 'CPD_PRODUCT_VERSION') }}"
+
+cpd_wml_dictionary:
```
Comment (whitfiea) [inline ibm/mas_devops/roles/suite_app_config/vars/predict.yml:17]: this is not used anywhere? Plus I am not sure what it is doing

File: ibm/mas_devops/roles/suite_app_config/vars/predict.yml [hunk 6]
```diff
@@ -5,3 +5,15 @@ mas_app_ws_kind: PredictWorkspace
...
 mas_app_cfg_delay: 120
 mas_app_cfg_retries: 30
+
+mas_app_settings_wml_version: "{{ lookup('env', 'WML_VERSION') | default('4.5', true) }}"
+mas_app_settings_watsonstudio_project_id: "{{ lookup('env', 'CPD_PREDICT_WS_PROJECT') | default('', true) }}"
+mas_app_settings_imagePullPolicy: "{{ lookup('env', 'MAS_IMAGE_PULL_POLICY') | default('IfNotPresent', true) }}"
+
+# CPD version is required by predict to identify what version of WML must be used to activate predict
+# if this information is used by more than predict, it should be moved to defaults/main.yml of the role
+cpd_product_version: "{{ lookup('env', 'CPD_PRODUCT_VERSION') }}"
```
Comment (whitfiea) [inline ibm/mas_devops/roles/suite_app_config/vars/predict.yml:15]: this is already in the defaults/main.yaml (line 24) is there a reason for it to be here?

**Review by whitfiea — CHANGES_REQUESTED** [review]
A couple more comments added. Also as this is technically removing function (the setting of the deployment size) we should probably make this a major change (unless this never worked previously). Can you also update the PR title to be more descriptive as OM 1.7 doesn't mean much. @durera any comment about the removal of the deployment size and major vs minor?

**Comment by ctrafa** [comment]
this pr has no point anymore. Closing it!

**Pull request closed without merging**